src/Microsoft.DotNet.Tools.Publish/PublishCommand.cs

```diff
@@ -14,124 +14,74 @@
-            var output = app.Option("-o|--output <OUTPUT_PATH>", "Path in which to publish the app", CommandOptionType.SingleValue);
-            var configuration = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
-            var project = app.Argument("<PROJECT>", "The project to publish, defaults to the current directory. Can be a path to a project.json or a project directory");
+        public int NumberOfFailedToPublishProjects;
 
-            app.OnExecute(() =>
+        public bool TryPrepareForPublish()
+        {
+            if (Framework != null)
             {
-                NuGetFramework nugetframework = null;
-
-                if (framework.HasValue())
-                {
-                    nugetframework = NuGetFramework.Parse(framework.Value());
-
-                    if (nugetframework.IsUnsupported)
-                    {
-                        Reporter.Output.WriteLine($"Unsupported framework {framework.Value()}.".Red());
-                        return 1;
-                    }
-                }
-
-                // TODO: Remove this once xplat publish is enabled.
-                if (!runtime.HasValue())
-                {
-                    runtime.Values.Add(RuntimeIdentifier.Current);
-                }
+                NugetFramework = NuGetFramework.Parse(Framework);
 
-                // Locate the project and get the name and full path
-                var path = project.Value;
-                if (string.IsNullOrEmpty(path))
+                if (NugetFramework.IsUnsupported)
                 {
-                    path = Directory.GetCurrentDirectory();
-                }
-
-                var projectContexts = ProjectContext.CreateContextForEachTarget(path);
-                projectContexts = GetMatchingProjectContexts(projectContexts, nugetframework, runtime.Value());
-
-                if (projectContexts.Count() == 0)
-                {
-                    string errMsg = $"'{project.Value}' cannot be published";
-                    if (framework.HasValue() || runtime.HasValue())
-                    {
-                        errMsg += $" for '{framework.Value()}' '{runtime.Value()}'";
-                    }
-
-                    Reporter.Output.WriteLine(errMsg.Red());
-                    return 1;
-                }
-
-                int result = 0;
-                foreach (var projectContext in projectContexts)
-                {
-                    result += Publish(projectContext, output.Value(), configuration.Value() ?? Constants.DefaultConfiguration);
+                    Reporter.Output.WriteLine($"Unsupported framework {Framework}.".Red());
+                    return false;
                 }
+            }
 
-                return result;
-            });
+            ProjectRidFxPermutations = ProjectContext.CreateContextForEachTarget(ProjectPath);
```

Comment:
What is so bad about referring to it as a ProjectContext? It encapsulates more than just the RID Fx pairing.